src/databricks/labs/ucx/source_code/linters/python_infer.py

```diff
@@ -0,0 +1,221 @@
+from __future__ import annotations
```

Comment:
don't mix refactoring and new features in the same PR. it's completely unclear to me what changed in these changes and what's the new code. please revert the introduction of this module in this PR.

Reply:
this was moved from python_ast to work around cyclic imports